src/extensions/extension-discovery.ts

```diff
@@ -347,29 +337,40 @@ export class ExtensionDiscovery {
         isEnabled
       };
     } catch (error) {
-      logger.error(`${logModule}: can't install extension at ${manifestPath}: ${error}`, { manifestJson });
+      logger.error(`${logModule}: can't load extension manifest at ${manifestPath}: ${error}`, { manifestJson });
 
       return null;
     }
   }
 
-  async loadExtensions(): Promise<Map<LensExtensionId, InstalledExtension>> {
+  async ensureExtensions(): Promise<Map<LensExtensionId, InstalledExtension>> {
     const bundledExtensions = await this.loadBundledExtensions();
 
-    await this.installPackages(); // install in-tree as a separate step
-    const localExtensions = await this.loadFromFolder(this.localFolderPath);
+    await this.installPackages(this.packageJsonPath, bundledExtensions);
 
-    await this.installPackages();
-    const extensions = bundledExtensions.concat(localExtensions);
+    const userExtensions = await this.loadFromFolder(this.localFolderPath);
```

Comment:
Would it be good, in a future PR, to remove the dependence on this folder so that we can no longer use explicit symlinks and instead have a list of "dev" extensions to just `require`?

Reply:
Yes, that would be good improvement (but cannot be done in a patch release).

Reply:
okay sounds good. Maybe 4.1?

Reply:
4.1 is already quite loaded but we can try.